File: examples/python/simple_test.py
#!/usr/bin/env python3
"""
Simple test script for debugging
"""

def main():
    """Main function for testing debugger."""
    a = 1
    b = 2
    print(f"Before swap: a={a}, b={b}")
    a, b = b, a  # We'll set a breakpoint on this line
    print(f"After swap: a={a}, b={b}")

if __name__ == "__main__":
    main()

File: examples/dotnet/SimpleTest.cs
using System;

namespace DebugTest
{
    class Program
    {
        static void Main(string[] args)
        {
            int a = 1;
            int b = 2;
            Console.WriteLine($"Before swap: a={a}, b={b}");
            int temp = a;  // Set breakpoint here (line 12)
            a = b;
            b = temp;
            Console.WriteLine($"After swap: a={a}, b={b}");

            int result = Add(3, 4);
            Console.WriteLine($"Add(3, 4) = {result}");
        }

        static int Add(int x, int y)
        {
            int sum = x + y;  // Set breakpoint here (line 22) for step-into test
            return sum;
        }
    }
}
